docs/source/federation-spec.mdx

```diff
@@ -0,0 +1,625 @@
+
+### `type _Service`
+
+This object type must have an `sdl: String!` field, which returns the SDL of the subgraph schema as a string.
+
+The returned schema string _must_ include all uses of federation-specific directives (`@key`, `@requires`, etc.), and it _must not_ include any definitions from the [subgraph schema specification](#subgraph-schema-specification).
+
+For details, see [Enhanced introspection with `Query._service`](#enhanced-introspection-with-query_service).
+
+### `union _Entity`
+
+**⚠️ This union type is generated dynamically based on the input subgraph schema!**
+
+This union's possible types must include all entities that the subgraph defines. It's the return type of the `Query._entities` field, which the graph router uses to directly access a subgraph's entity fields.
+
+For details, see [Defining the `_Entity` union](#defining-the-_entity-union).
+
+### `scalar _Any`
+
+This scalar is the type used for entity **representations** that the graph router passes to the `Query._entities` field. An `_Any` scalar is validated by matching its `__typename` and `@key` fields against entities defined in the subgraph schema.
+
+### `scalar FieldSet`
+
+This string-serialized scalar represents a set of fields that's passed to a federated directive, such as `@requires` or `@provides`.
+
+Grammatically, a `FieldSet` is a [selection set](http://spec.graphql.org/draft/#sec-Selection-Sets) minus the outermost curly braces. It can represent a single field (`"upc"`), multiple fields (`"id countryCode"`), and even nested selection sets (`"id organization { id }"`).
+
+### == Added directives ==
+
+### `@external`
+
+```graphql
+directive @external on FIELD_DEFINITION
+```
+
+The `@external` directive indicates that a particular subgraph _cannot_ resolve a particular entity field. This enables a subgraph to reference a _different_ subgraph's field _without_ also resolving that field.
+
+The `@external` directive is always used in combination with another directive that references fields, such as `@requires` or `@provides`:
+
+```graphql title="Shipping subgraph"
+type Product @key(fields: "upc") {
+  upc: String!
+  weight: Int! @external
+  shippingCost: Int! @requires("weight")
+}
+```
+
+In the example above, the Shipping subgraph adds the `shippingCost` field to the `Product` entity. This field requires the value of the `weight` field, which _cannot_ be resolved by the Shipping subgraph. Therefore, the Shipping subgraph defines the `weight` field but also marks it as `@external`.
+
+### `@inaccessible`
+
+```graphql
+directive @inaccessible(from: String!) on FIELD_DEFINITION | INTERFACE | OBJECT | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION
+```
+
+The `@inaccessible` directive indicates that a location in the subgraph schema should _not_ be included in the [API schema](./federated-types/overview/#api-schema) that clients use to execute operations against the graph router.
```

Comment:
lets add a primary use case -> it can be used for defining private keys that cannot be queried by the api schema but will be used by gateway for composition

Reply:
This is a cool and powerful use case that is very important and that I had never considered 😮